ledger/participant-integration-api/src/main/scala/platform/apiserver/services/tracking/TrackerImpl.scala

```diff
@@ -95,24 +93,10 @@ private[services] object TrackerImpl {
       )
       .viaMat(tracker)(Keep.both)
       .toMat(Sink.foreach { case Ctx(promise, result, _) =>
-        result match {
-          case compl @ Completion(_, Some(Status(Code.OK.value, _, _, _)), _) =>
-            logger.trace("Completing promise with success")
-            promise.trySuccess(compl)
-          case Completion(_, statusO, _) =>
-            val status = statusO
-              .map(status =>
-                GrpcStatus
-                  .fromCodeValue(status.code)
-                  .withDescription(status.message)
-              )
-              .getOrElse(
-                GrpcStatus.INTERNAL
-                  .withDescription("Missing status in completion response.")
-              )
-
-            logger.trace(s"Completing promise with failure: $status")
-            promise.tryFailure(status.asException())
+        logger.trace("Completing promise")
+        val didCompletePromise = promise.trySuccess(result)
+        if (!didCompletePromise) {
+          logger.trace("Failed to complete promise")
```

Comment:
I'd perhaps change it to:

- Log only afterwards, both in the `true` and `false` cases
- Expand a bit the log message to give more context about what's happening